base/docs/helpdb/Base.jl

```diff
@@ -881,7 +884,7 @@ trunc
 """
     unsafe_convert(T,x)
 
-Convert `x` to a value of type `T`
+Convert `x` to a C argument of type `T` where `x` is the return value of `cconvert(T, ...)`.
```

Comment:
>  It's not allowed to call `unsafe_convert` with `x` being anything that's not the return value of `cconvert`.

Of course, in special case you can do that if you know what the `cconvert` would return or what `unsafe_convert` would accept. But that's an implementation detail and varies for different types. There are some of them in base that might allow this (mainly because there are actually (incorrect) use in packages...) so those can be documented but that's not true in general and should not be in the generic doc string.